udata/core/user/api.py

```diff
@@ -205,25 +210,67 @@ def delete(self):
 
 
 @ns.route('/', endpoint='users')
-@api.doc(get={
-    'id': 'list_users',
-    'model': user_page_fields,
-    'parser': search_parser})
-@api.doc(post={'id': 'create_user', 'model': user_fields})
-class UserListAPI(ModelListAPI):
+class UserListAPI(API):
     model = User
     fields = user_fields
     form = UserProfileForm
     search_adapter = UserSearch
 
+    @api.doc('list_users')
+    @api.expect(search_parser)
+    @api.marshal_with(user_page_fields)
+    def get(self):
+        '''List all users'''
+        return search.query(UserSearch, **multi_to_dict(request.args))
+
+    @api.secure
+    @api.doc('create_user')
+    @api.expect(user_fields)
+    @api.marshal_with(user_fields, code=201)
+    @api.response(400, 'Validation error')
+    def post(self):
+        '''Create a new object'''
+        form = api.validate(UserProfileForm)
+        user = form.save()
+        return user, 201
+
 
 @ns.route('/<user:user>/', endpoint='user')
-@api.doc(model=user_fields, get={'id': 'get_user'})
-@api.doc(put={'id': 'update_user', 'body': user_fields})
-class UserAPI(ModelAPI):
-    model = User
-    fields = user_fields
-    form = UserProfileForm
+@api.response(404, 'User not found')
+@api.response(410, 'User has been deleted')
+class UserAPI(API):
+    @api.doc('get_user')
+    @api.marshal_with(user_fields)
+    def get(self, user):
+        '''Get an user given its identifier'''
```

Comment:
_Delete a user_ (_u_ being pronounced 'you' there's no 'n')
This applies for other '_an user_' comments in this PR